pkg/api/types.go

```diff
@@ -256,6 +258,19 @@ type SecretVolumeSource struct {
 	Target ObjectReference `json:"target"`
 }
 
+// NFS represents and NFS Mount that lasts the lifetime of a pod
+type NFSVolumeSource struct {
+	// Server is the hostname or ip address of the NFS server
+	Server string `json:"server"`
+
+	// SourcePath is the exported NFS share
+	SourcePath string `json:"sourcePath"`
+
+	// MountOptions is a comma delimited list of mount options. The available options are listed on the mount man page as arguments to -o.
+	// As an example: noatime,nodev,nosuid,ro
+	MountOptions string `json:"mountOptions"`
```

Comment:
Well, the thing is this is using the mount command underneath, so it's not really the same thing. And having a variable that can take all the mount options then combine that with a readOnly option seems superfluous. I think you made the note here as well about turning this into a []string and I mentioned I didn't understand why we would do that if it just needs to be flattened into string argument for value of the mount -o flag.